src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -1250,6 +1252,19 @@ private void ParseModifiers(SyntaxListBuilder tokens, bool forAccessors)
                         modTok = CheckFeatureAvailability(modTok, MessageID.IDS_FeatureAsync);
                         break;
 
+                    case DeclarationModifiers.Required:
+                        // In C# 11, required in a modifier position is always a keyword if not escaped. Otherwise, we reuse the async detection
+                        // machinery to make a conservative guess as to whether the user meant required to be a keyword, so that they get a good langver
+                        // diagnostic and all the machinery to upgrade their project kicks in.
+                        if (!IsFeatureEnabled(MessageID.IDS_FeatureRequiredMembers) && !ShouldAsyncOrRequiredBeTreatedAsModifier(parsingStatementNotDeclaration: false))
```

Comment:
I would have only expected the LangVer check, not the additional `ShouldAsyncOrRequiredBeTreatedAsModifier` condition.

Reply:
The reason I did this because I'm concerned about the langversion upgrade story. I'm not going as far out of my way as I would if I were trying to support all scenarios, but otherwise there would be _no_ langversion errors when a user on a lower language version tries to use the feature at all. We'd just never parse it as a `required` modifier at all, and I don't think that's an acceptable user experience.

Reply:
Could you add a test for `required required;` (a top-level statement declaring local of type `required` and named `required`) in C# 10? ([sharplab](https://sharplab.io/#v2:EYLgtghglgdgNAFxBAzguATEBqAPgJwFMBHAVyiIwAIiyLCMBuAWACg2ABAJhpPMqoBvKgF8gA==) shows the expected parse tree)

Reply:
Done. Good suggestion, it revealed a bug.